src/canvasPanelComponents/slideshow/src/components/CanvasNavigation/CanvasNavigation.js

```diff
@@ -1,9 +1,34 @@
-import React, { Component } from 'react';
+import React, { Component, useEffect } from 'react';
 import { withBemClass } from '@canvas-panel/core';
+import withLocation from '../withLocation/withLocation';
 
 import './CanvasNavigation.scss';
 
 class CanvasNavigation extends Component {
+  componentDidMount() {
+    if (this.props.addressable && this.props.hash) {
+      let slideId =
+        this.props.hash.includes('slide') && this.props.hash.split('=')[1];
+      if (
+        !this.props.hash ||
+        !slideId ||
+        slideId < 0 ||
+        !Number.isInteger(parseInt(slideId)) ||
+        slideId >= this.props.canvasList.length
+      ) {
+        this.goToSlide(0);
+        return;
+      } else {
+        this.props.goToRange(parseInt(slideId));
+      }
+    }
+  }
+  goToSlide = index => {
+    if (this.props.addressable) {
+      document.location.hash = '#?slide=' + index;
```

Comment:
Do you think we could add one more prop to the base component (like `id="1234"`) that could also be checked against the hash? (`#?id=1234&slide=1234`) I think it's a common case to have one or more of these on the page.